File: src/netcode/emscripten_serverinfo.js
mergeInto(LibraryManager.library, {
  SRB2_ServerInfoResponse: function (
    serverName,
    map,
    mapTitle,
    ingame_players
  ) {
    if (window.SRB2_ServerInfoResponse) {
      return window.SRB2_ServerInfoResponse({
        name: UTF8ToString(serverName),
        map: UTF8ToString(map),
        mapTitle: UTF8ToString(mapTitle),
        ingamePlayers: ingame_players,
      });
    }
    return 1;
  },
});

File: src/netcode/emscripten_serverinfo.c
//Used to pass in server info for the relay server in SRB2 web.
//Simple c script created by gvbvdxx.

#ifdef EMSCRIPTEN
#include <emscripten.h>

#include "../g_game.h"
#include "../doomstat.h"
#include "../doomdef.h"
#include "../command.h"
#include "../i_threads.h"
#include "../m_menu.h"
#include "../z_zone.h"
#include "mserv.h"
#include "client_connection.h"

extern void SRB2_ServerInfoResponse(
    char *name,
    char *map,
    char *map_title
);

EMSCRIPTEN_KEEPALIVE
void SRB2_GetServerInfo(void)
{
    SRB2_ServerInfoResponse(
        (char *)cv_servername.string,
        G_BuildMapName(gamemap),
        G_BuildMapTitle(gamemap)
    );
}
#endif